### ISSUE TITLE
Inheritance of @(unordered) by group rules

### ISSUE THREAD

**Issue by anewton1998** [arineng/jcr#1]
We need to figure out what to do about gropus in unordered arrays:

```
@(unordered) [ (foo,bar), baz ]
```

Does that mean `(foo,bar)` is evaluated unordered too?

**Comment by codalogic** [comment]
First thought is that given that you can't revert back to ordered something that has inherited unordered, maybe the inner one has to be ordered, and to get it all unordered you have to do:

```
@(unordered) [ @(unordered) (foo,bar), baz ]
```

That looks very messy though.  So perhaps the whole thing should be considered unordered.  If you want the (foo,bar) bit to be ordered you have to do:

```
@(unordered) [ foo-bar, baz ]
foo-bar ( foo, bar )
```

As rule foo-bar doesn't have the unordered annotation it is ordered.

That said, it may be worth thinking about how it would be coded up!

**Comment by anewton1998** [comment]
So you are suggesting that we allow @(unordered) on groups too?  I was thinking groups would inherit the behavior, but having it explicitly stated is an interesting idea.

**Comment by codalogic** [comment]
In 4.13 of the draft it says:

```
Additionally, annotations may not decorate references to rules
(target rule names).  The following is not correct:

    [ * @(unordered) bar ]
```

That makes sense to me.  (There is only one definition of 'bar' and you can't modify it on a case-by-case basis.)  I had also taken that to mean that more 'outer' annotations wouldn't affect the named rule either.

So I guess I am saying allow annotations on groups!

**Comment by anewton1998** [comment]
I hadn't thought of that, but I think it would be good to try. Let me see if I can get my validator to do it to see how hard such a thing would be. At present, it is buggy with groups and arrays. There's some stuff I need to fix.

**Comment by anewton1998** [comment]
So I have implemented the @(unordered) annotations on groups in my validator, and I have to say that it is just nuts. For validation to go from ordered to unordered is easy, but switching back to ordered gets problematic. Where is the ordered validation to pick up? I picked the highest index of the match from the unordered group, but what if there are gaps? It would not be obvious what was going on.

In my opinion, switching from ordered to unordered or vice-versa would be a seldom used feature. But it would be nasty to implement and confusing in practice. I think groups should always inherit the behavior of their parent array. If somebody needs something fancy like this, they need to a) rethink what they are doing, and b) write custom code.
